pages/en/developer/querying-from-your-app.mdx

```diff
@@ -2,135 +2,572 @@
 
-```javascript
-const tokensQuery = `
-  query($first: Int, $orderBy: BigInt, $orderDirection: String) {
-    tokens(
-      first: $first, orderBy: $orderBy, orderDirection: $orderDirection
-    ) {
+- it makes it **harder to understand** the query as a whole
+- developers are **responsible for safely sanitizing the string interpolation**
+- not sending the values of the variables as part of the request parameters **prevent possible caching on server-side**
+- it **prevents tools from statically analyzing the query** (ex: Linter, or type generations tools)
+
+For this reason, it is recommended to always write queries as static strings:
+
+```tsx
+import { execute } from "your-favorite-graphql-client"
+
+const id = params.id
+const query = `
+query GetToken($id: ID!) {
+  token(id: $id) {
+    id
+    owner
+  }
+}
+`
+
+const result = await execute(
+  query, 
+  {
+    variables: {
       id
-      tokenID
-      contentURI
-      metadataURI
     }
   }
+)
+```
+
+Doing so brings **many advantages**:
+
+- **Easy to read and maintain** queries
+- The GraphQL **server handles variables sanitization**
+- **Variables can be cached** at server-level
+- **Queries can be statically analyzed by tools** (more on this in the following sections)
+
+**Note: How to include fields conditionally in static queries**
+
+We might want to include the `owner` field only on a particular condition.
+
+For this, we can leverage the `@include(if:...)` directive as follows:
+
+```tsx
+import { execute } from "your-favorite-graphql-client"
+
+const id = params.id
+const query = `
+query GetToken($id: ID!, $includeOwner: Boolean) {
+  token(id: $id) {
+    id
+    owner @include(if: $includeOwner)
+  }
+}
 `
 
-client
-  .query({
-    query: gql(tokensQuery),
+const result = await execute(
+  query, 
+  {
     variables: {
-      first: 10,
-      orderBy: 'createdAtTimestamp',
-      orderDirection: 'desc',
-    },
-  })
-  .then((data) => console.log('Subgraph data: ', data))
-  .catch((err) => {
-    console.log('Error fetching data: ', err)
-  })
+      id,
+      includeOwner: true
+    }
+  }
+)
 ```
 
-### URQL
+Note: The opposite directive is `@skip(if: ...)`.
+
+<br />
 
-Another option is [URQL](https://formidable.com/open-source/urql/), a somewhat lighter weight GraphQL client library.
+### Performance tips
 
-Let's look at how fetch data from a subgraph with URQL in a web project.
+**"Ask for what you want"**
 
-First, install `urql` and `graphql`:
+GraphQL became famous for its “Ask for what you want” tagline.
 
-```sh
-npm install urql graphql
+For this reason, there is no way, in GraphQL, to get all available fields without having to list them individually.
+
+When querying GraphQL APIs, always think of querying only the fields that will be actually used.
+
+Since some fields might be computed or proxied on the server-side, fetching fewer fields can make a big difference!
```

Comment:
We don't have computed fields (yet) A much bigger concern is overfetching of collections since the default value for `first` is `100`. It would be good to add a paragraph about that, something like:

A common cause of overfetching are collections of entities. By default, queries will fetch 100 entities in a collection, which is usually much more than what will actually be used, e.g., for display to the user. Queries should therefore almost always set `first` explicitly, and make sure they only fetch as many entities as they actually need. This applies not just to top-level collections in a query, but even more so to nested collections of entities. For example, in the query
```graphql
query listTokens {
  tokens {
    id
    transactions { id }
  }
}
```
the response could contain 100 transactions for each of the 100 tokens in the response. If the application only needs 10 transactions, the query should set `first: 10` explicitly on the `transactions` field.

Reply:
Thanks for the review @lutter!

I added your suggestion via b656828.